test/unit/three_scale/analytics/user_tracking_test.rb

```diff
@@ -8,125 +8,176 @@ def setup
-    refute tracking.track('Event')
-  end
+      tracking = UserTracking.new(@user)
+      assert tracking.group
 
-  def test_group
-    tracking = UserTracking.new(nil)
-    refute tracking.group
+      @user.account.provider = false
+      refute tracking.group
+    end
 
-    tracking = UserTracking.new(@user)
-    assert tracking.group
+    def test_group_traits
+      tracking = UserTracking.new(nil)
+      assert_equal Hash.new, tracking.group_traits
 
-    @user.account.provider = false
-    refute tracking.group
-  end
+      tracking = UserTracking.new(@user)
+      @user.account.stubs(bought_plan: Plan.new(name: 'fooplan', cost_per_month: 42), state: 'suspended')
 
-  def test_group_traits
-    tracking = UserTracking.new(nil)
-    assert_equal Hash.new, tracking.group_traits
+      assert_equal({name: @user.account.name,
+                    plan: 'fooplan',
+                    monthly_spend: 42.0,
+                    license_mrr: 42.0,
+                    state: 'suspended',
+                   }, tracking.group_traits)
+    end
 
-    tracking = UserTracking.new(@user)
-    @user.account.stubs(bought_plan: Plan.new(name: 'fooplan', cost_per_month: 42), state: 'suspended')
+    def test_can_send?
+      refute UserTracking.new(nil).can_send?
+      assert UserTracking.new(@user).can_send?
+    end
 
-    assert_equal({name: @user.account.name,
-                  plan: 'fooplan',
-                  monthly_spend: 42.0,
-                  license_mrr: 42.0,
-                  state: 'suspended',
-                 }, tracking.group_traits)
-  end
+    def test_only_track_providers
+      @user.account.provider = false
+      refute UserTracking.new(@user).can_send?
+    end
 
-  def test_can_send?
-    refute UserTracking.new(nil).can_send?
-    assert UserTracking.new(@user).can_send?
+    def test_do_not_track_impersonation_admins
+      @user.username = ThreeScale.config.impersonation_admin['username']
+      refute UserTracking.new(@user).can_send?
+    end
   end
 
-  def test_only_track_providers
-    @user.account.provider = false
-    refute UserTracking.new(@user).can_send?
+  class SegmentDisabled < UserTrackingTest
+    def setup
+      super
+      Features::SegmentConfig.stubs(enabled?: false)
+    end
+
+    def test_does_not_track
+      expect_not_contact_segment(:track)
+
+      UserTracking.new(@user).track('foo')
+    end
+
+    def test_does_not_flush
+      expect_not_contact_segment(:flush)
+
+      UserTracking.new(@user).flush
+    end
+
+    def test_does_not_identify
+      expect_not_contact_segment(:identify)
+
+      UserTracking.new(@user).identify
+    end
+
+    def test_does_not_group
+      expect_not_contact_segment(:group)
+
+      UserTracking.new(@user).group
+    end
+
+    private
+
+    def expect_not_contact_segment(action_name)
+      UserTracking::TrackingAdapter::NullAdapter.any_instance.expects(action_name).once
+      Segment::Analytics.any_instance.expects(action_name).never
```

Comment:
We all do the same despite knowing that `expects(...).never` tends to lead to outdated tests. However, it makes me think whether we should add a log message when `UserTracking` methods are used and segment is disabled, and then assert based on that log message instead of using `never`. WDYT?